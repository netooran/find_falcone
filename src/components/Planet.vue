<template>
  <div class="planet">
    <i class="fas fa-globe-africa fa-8x"></i>
    <div class="planet-info">
      <h1>{{ planet.name }}</h1>
      <h4>Distance: {{ planet.distance }} au</h4>
    </div>
  </div>
</template>

<script>
export default {
  name: "Planet",
  props: {
    planet: { name: String, distance: Number }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h4 {
  color: darkgrey;
}

.planet {
  padding: 30px;
  border-radius: 10px;
  box-shadow: 1px 1px 1px 1px #ccc;
}

.planet-info {
  margin-top: 8%;
}

.planet-info > * {
  margin: 0%;
}

.planet:hover {
  box-shadow: 1px 1px 3px 4px rgb(155, 154, 154);
}
</style>
