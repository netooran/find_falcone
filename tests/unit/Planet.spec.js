import { shallowMount } from "@vue/test-utils";
import Planet from "@/components/Planet.vue";

describe("Planet", () => {
  let wrapper, planet;

  beforeEach(() => {
    planet = { name: "Toliman", distance: 437 };
    wrapper = shallowMount(Planet, { propsData: { planet } });
  });

  it("renders name and distance when passed", () => {
    expect(wrapper.text()).toMatch(planet.name);
    expect(wrapper.text()).toMatch(`${planet.distance} au`);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
