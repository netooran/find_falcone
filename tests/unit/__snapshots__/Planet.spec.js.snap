// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet has the expected html structure 1`] = `
<div
  class="planet"
>
  <i
    class="fas fa-globe-africa fa-8x"
  />
   
  <div
    class="planet-info"
  >
    <h1>
      Toliman
    </h1>
     
    <h4>
      Distance: 437 au
    </h4>
  </div>
</div>
`;
